Src/HALALMock/Services/Communication/SPI/SPI.cpp

```diff
@@ -189,27 +164,13 @@ bool SPI::transmit_and_receive(uint8_t id, span<uint8_t> command_data,
         return false;
     }
 
-    SPI::Instance* spi = SPI::registered_spi[id];
-
-    HAL_StatusTypeDef errorcode =
-        HAL_SPI_TransmitReceive_DMA(spi->hspi, command_data.data(),
-                                    receive_data.data(), command_data.size());
-    turn_on_chip_select(spi);
-    switch (errorcode) {
-        case HAL_OK:
-            return true;
-            break;
-        case HAL_BUSY:
-            return false;
-            break;
-        default:
-            ErrorHandler(
-                "Error while transmiting and receiving with spi DMA. Errorcode "
-                "%u",
-                (uint8_t)errorcode);
-            return false;
-            break;
+    if(SPI::transmit(id, command_data) && SPI::receive(id, receive_data)) {
+        HAL_SPI_TxRxCpltCallback(SPI::registered_spi[id]->hspi);
```

Comment:
should this be here? it is introducing the HAL Typedef, I don't know the policy on this

Reply:
Other posibility is renaming callback name, but, although this function is only used by HAL, is a public method and I would be changing a public interface.

Reply:
I think it is possible that user code uses this callback underneath, we can leave it like is for now and come back later if it doesn't compile